src/debugging/python/PythonDebugHelper.ts

```diff
@@ -0,0 +1,196 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as vscode from "vscode";
+import { PythonExtensionHelper } from "../../tasks/python/PythonExtensionHelper";
+import { PythonDefaultDebugPort, PythonProjectType } from '../../utils/pythonUtils';
+import ChildProcessProvider from "../coreclr/ChildProcessProvider";
+import CliDockerClient from "../coreclr/CliDockerClient";
+import LocalOSProvider from "../coreclr/LocalOSProvider";
+import { DebugHelper, DockerDebugContext, DockerDebugScaffoldContext, inferContainerName, ResolvedDebugConfiguration, resolveDockerServerReadyAction } from '../DebugHelper';
+import { DockerDebugConfigurationBase } from "../DockerDebugConfigurationBase";
+import { DockerDebugConfiguration } from "../DockerDebugConfigurationProvider";
+import { PythonScaffoldingOptions } from '../DockerDebugScaffoldingProvider';
+
+export interface PythonPathMapping {
+    localRoot: string;
+    remoteRoot: string;
+}
+
+export interface PythonDebugOptions {
+    host?: string;
+    port?: number;
+    pathMappings?: PythonPathMapping[];
+    justMyCode?: boolean;
+    projectType?: PythonProjectType;
+    django?: boolean;
+    jinja?: boolean;
+}
+
+export interface PythonDockerDebugConfiguration extends DockerDebugConfigurationBase {
+    python?: PythonDebugOptions;
+}
+
+export class PythonDebugHelper implements DebugHelper {
+    public constructor(
+        private readonly cliDockerClient: CliDockerClient,
+        private readonly localOsProvider: LocalOSProvider) {
+    }
+
+    public async provideDebugConfigurations(context: DockerDebugScaffoldContext, options?: PythonScaffoldingOptions): Promise<DockerDebugConfiguration[]> {
+        const configs : DockerDebugConfiguration[] =
+        [{
+            name: "Docker Python Launch",
+            type: "docker",
+            request: "launch",
+            preLaunchTask: "docker-run: debug",
+            python: {
+                pathMappings: [
+                    {
+                        /* eslint-disable-next-line no-template-curly-in-string */
+                        localRoot: "${workspaceFolder}",
+                        remoteRoot: "/app"
+                    }
+                ],
+                projectType: options.projectType
+            }
+        }];
+
+        // If we generated compose files, then we should generate a Python attach configuration.
+        if (context.generateComposeTask) {
+            configs.push(
+                {
+                    name: "Python: Remote Attach",
```

Comment:
We use colon here in the name but not in the other configuration?

Reply:
The thinking is that for the Docker configuration, the vote was not to use the colon. However, for the Python attach one, that is how it gets generated by default from the extension so we thought we'd use their defaults.

Reply:
Yeah, I get that and, if we were generating them one-by-one, that would be reasonable.  However, we generate these at the *same* time and, if they're the only ones the user ever adds to `launch.json`, it just looks odd to me.